src/preprocess/links.rs

```diff
@@ -58,8 +59,18 @@ fn replace_all<P: AsRef<Path>>(s: &str, path: P) -> String {
 
         match playpen.render_with_path(&path) {
             Ok(new_content) => {
-                replaced.push_str(&new_content);
-                previous_end_index = playpen.end_index;
+                if depth < 200 {
+                    replaced.push_str(&replace_all(&new_content, path, &source, depth + 1));
+                    previous_end_index = playpen.end_index;
+                }
+                else {
+                    if let Some(source_path) = source.as_os_str().to_str() {
```

Comment:
You can actually remove the need for the `if let Some(...)` and two branches by using `source.display()` in the `error!()` message.